src/lib/OpenCvProvider.js

```diff
@@ -6,53 +6,39 @@ const { Consumer: OpenCvConsumer, Provider } = OpenCvContext
 
 export { OpenCvConsumer, OpenCvContext }
 
-const scriptId = 'opencv-react'
-const moduleConfig = {
-  wasmBinaryFile: 'opencv_js.wasm',
-  usingWasm: true
-}
-
-export const OpenCvProvider = ({ openCvPath, children, onLoad }) => {
-  const [loaded, setLoaded] = React.useState(false)
-
-  const handleOnLoad = React.useCallback(() => {
-    if (onLoad) {
-      onLoad(window.cv)
-    }
-    setLoaded(true)
-  }, [])
+export const OpenCvProvider = ({ openCvVersion = '4.5.5', openCvPath = '', children }) => {
+  const [cvInstance, setCvInstance] = React.useState({ loaded: false, cv: undefined });
 
   React.useEffect(() => {
+    const scriptId = 'opencv-react'
     if (document.getElementById(scriptId) || window.cv) {
-      setLoaded(true)
       return
     }
 
+    window.Module = window.Module || {
+      wasmBinaryFile: 'opencv_js.wasm',
+      usingWasm: true,
+      onRuntimeInitialized: () => {
+        console.log('opencv loaded', window.cv)
+        setCvInstance({ loaded: true, cv: window.cv });
+      }
+    }
+
     // https://docs.opencv.org/3.4/dc/de6/tutorial_js_nodejs.html
     // https://medium.com/code-divoire/integrating-opencv-js-with-an-angular-application-20ae11c7e217
     // https://stackoverflow.com/questions/56671436/cv-mat-is-not-a-constructor-opencv
```

Comment:
these comments should be placed above window.module, because they explain why we need to do such a thing

Reply:
good, will put them there.